### PR TITLE
Fix test timeouts

### PR DESCRIPTION
**Pull request by jswoods**
Fixes a number of issues including the timeouts in `waitForPod()` when running tests, race condition between the pod certificate and the pod creation, and PVC support for migrating from ephemeral to PVCs.

### CODE REVIEW

File: controllers/humiocluster_tls.go
Comment (SaaldjorMike) [inline controllers/humiocluster_tls.go:245]: Perhaps we could do the same json marshal here before calculating the hash?

File: controllers/humiocluster_tls.go
Comment (SaaldjorMike) [inline controllers/humiocluster_tls.go:226]: I don't think the method name is good at conveying what it is doing. To me this "count" suffix makes me think that it just returns the number of certificates we have right now, but doesn't touch the resources. This looks to not be true as it does trigger updates of certificates if their hash differs.

Perhaps it would be fine if we'd just add some godoc for it, but I think we should see if we can find a more suitable method name.

**Review by SaaldjorMike — APPROVED** [review]
Added a few minor comments, but LGTM. I'll leave it up to you to decide if you want to fix those before merging.

File: controllers/humiocluster_tls.go
Comment (jswoods) [reply controllers/humiocluster_tls.go:226]: Right, this was a poor name choice.

File: controllers/humiocluster_tls.go
Comment (jswoods) [reply controllers/humiocluster_tls.go:245]: Yes.

**Pull request merged**
